### ISSUE TITLE
Add support for IN operator

### ISSUE THREAD

**Issue by timpengu** [OData/odata.net#1147]
The [OData Version 4.01](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html) specification adds support for [IN expressions](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_in) such as:

`http://host/service/Products?$filter=Name in ('Milk', 'Cheese')`

ODataLib should support this operator syntax.

### Assemblies affected

ODataLib 7.4.4

### Additional details

Although this can currently be be achieved using multiple OR clauses, some benefits of IN expressions are:
* much shorter query string when filtering on a large set of values (max URI length is limited)
* often simpler to translate to efficient queries on an underlying data source (e.g. I am translating OData requests to a MongoDB backend where a single $in query executes much faster than the equivalent $or queries).

**Comment by AlanWong-MS** [comment]
This one's in progress by me. :) It's currently in the design phase. I'll use this issue as a way to maintain updates. Thanks for opening it!

**Comment by NetTecture** [comment]
Has anything happened on this in the last 6 weeks? I consider this one of the most "wtf" shortcomings in the OData Specs (how the heck could they overlook that) which finally is closed - and am desperately waiting on being able to use it.

**Comment by NetTecture** [comment]
@AlanWong-MS Well, face it - OData (the protocol) is quite interesting in capabilities. Nice how it took so long actually for someone to realize that remove functions (even in REST shape) re worth nearly nothing unless you ass query and projection syntax to the definition, so you do not end up with dozens of nearly redundant methods ;) The reason I am so behind this is that next week I set the basics for a new project and start implementing the API There in dotnet core 2.1 odata - at least until I run into too many problems that don't get fixes to make sense.... which I di not expect ;) Good to know WebApi OData 7 is getting along.

**Comment by mikepizzo** [comment]
Yes, this is a feature that we are certainly interested in getting added -- thanks for all of your enthusiasm!

**Comment by evilB** [comment]
I'm looking forward to being able to use the in clause, currently running into maxnode isssues due to using multiple or sequences in the filter. 

i've subscribed to remain informed

**Comment by AlanWong-MS** [comment]
This feature has been merged into master via #1165. Our release with the feature is coming up :)

**Comment by timpengu** [comment]
Excellent, thanks Alan!  Looking forward to integrating this into our system, and persuading our clients to  rewrite their queries :)
